src/material/datepicker/date-range-input.scss

```diff
@@ -0,0 +1,91 @@
+@import '../core/style/vendor-prefixes';
+
+$mat-date-range-input-separator-spacing: 4px;
+$mat-date-range-input-part-max-width: calc(50% - #{$mat-date-range-input-separator-spacing});
+
+// Host of the date range input.
+.mat-date-range-input {
+  display: block;
+  width: 100%;
+}
+
+// Inner container that wraps around all the content.
+.mat-date-range-input-container {
+  display: flex;
+  align-items: center;
+}
+
+// Text shown between the two inputs.
+.mat-date-range-input-separator {
+  margin: 0 $mat-date-range-input-separator-spacing;
+}
+
+// Underlying input inside the range input.
+.mat-date-range-input-inner {
+  // Reset the input so it's just a transparent rectangle.
+  font: inherit;
+  background: transparent;
+  color: currentColor;
+  border: none;
+  outline: none;
+  padding: 0;
+  margin: 0;
+  vertical-align: bottom;
+  text-align: inherit;
+  -webkit-appearance: none;
+  width: 100%;
+
+  // Remove IE's default clear and reveal icons.
+  &::-ms-clear,
+  &::-ms-reveal {
+    display: none;
+  }
+}
+
+// We want the start input to be flush against the separator, no matter how much text it has, but
```

Comment:
Because it looks weird if you typed something shorter and then the space around the separator was uneven, e.g. `11/11     - 12/12`.

Reply:
What if we right-align the the input?

Reply:
Which one? The start date input or the date range input in general? I think we'd just the same problem but in a different place.

Reply:
I pulled down the PR and tried tweaking a few things locally. I concluded that everything I suggested looks worse than what you ultimately did. I have one more idea, though, which is to hide the end-date input placeholder once you start typing into the start-date input. I tried this out locally and I think it eliminates the appearance of the input getting pushed.

Reply:
Sure, that should be easy to do.

Reply:
We should also think about the edge cases related to this. e.g. what happens if I skip the start and type in the end input?

If we want to go with Jeremy's suggestion my mental model would be to treat them as a single placeholder, i.e. whenever one of the inputs has text in it both placeholders are hidden

Reply:
Good point that hiding it could introduce weird edge cases. Also if we hid only one of the fields, the user won't have a way of knowing that it's there.

Reply:
Agreed that we should treat them conceptually as one placeholder. Do you want to do that in this PR or in a follow-up?

Reply:
I'll do it in a follow-up. There should be some more things to clean up once this gets in.

Reply:
Please leave TODO's for anything that you will handle in a follow-up PR. I feel we've been getting a little lax with the "will handle in a follow-up PR" lately, and I'm worried that things might slip off our radar and never actually get done

Reply:
I ended up just doing it now. Can you take one more look @mmalerba? The way it behaves now is that it hides the placeholders when one of these criteria is true:

- The inputs are empty and they have focus.
- The start input is focused and the end input is empty.

Reply:
The way I was imagining it, the dash doesn't disappear. @jelbourn should take a look too, since he felt more strongly about the placeholder stuff

Reply:
The way I was reading it the dash should disappear, otherwise you'll see it jump to the beginning when you start typing.

Reply:
I've updated it to keep the separator visible while the start has a value.